### PR TITLE
Adding support for cross-instance PITR for managed instances

### PR DESCRIPTION
**Pull request by petrajkogit**
## Description

This change is adding cross-instance Point in Time restore for managed instances.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Sql/Commands.Sql/ManagedDatabase/Cmdlet/RestoreAzureSqlManagedDatabase.cs [hunk 1]
```diff
@@ -127,6 +127,20 @@ public class RestoreAzureRmSqlManagedDatabase
...
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The name of the target managed instance to restore to.")]
+        public string TargetManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the target resource group to restore to. If not specified, the target resource group is same as source
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The name of the target resource group to restore to.")]
```
Comment (jaredmoo) [inline src/ResourceManager/Sql/Commands.Sql/ManagedDatabase/Cmdlet/RestoreAzureSqlManagedDatabase.cs]: Please document behavior if unspecified, e.g. "The name of the target resource group to restore to. If not specified, the target resource group is the same as the source resource group."

File: src/ResourceManager/Sql/Commands.Sql/help/Restore-AzureRmSqlManagedDatabase.md [hunk 2]
```diff
@@ -198,6 +214,32 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -TargetManagedInstanceName
+The name of the target managed instance to restore to.
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/help/Restore-AzureRmSqlManagedDatabase.md]: Please add back the ```yaml tag.  It should be on a new line under the description.

File: src/ResourceManager/Sql/Commands.Sql/help/Restore-AzureRmSqlManagedDatabase.md [hunk 3]
```diff
@@ -198,6 +214,32 @@ Accept pipeline input: False
...
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -TargetResourceGroupName
+The name of the target resource group to restore to.
```
Comment (maddieclayton) [inline src/ResourceManager/Sql/Commands.Sql/help/Restore-AzureRmSqlManagedDatabase.md]: same

**Comment by maddieclayton** [comment]
@petrajkogit Can you link the design review for this change? If you haven't done one, please add a cmdlet design review here: https://github.com/Azure/azure-powershell-cmdlet-review-pr

**Comment by petrajkogit** [comment]
@maddieclayton : Here is the link to design review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/126

**Comment by maddieclayton** [comment]
@petrajkogit Please either update this PR or close if you aren't planning on updating soon.

**Comment by petrajkogit** [comment]
@maddieclayton I updated the json file after rerunning the tests. Sorry for the delay - we had a week long issue with the cluster I am running tests on. However, I see build is failing with something that is no longer in new code, looks like some caching issue. Can you take a look, please?

ManagedDatabase\Cmdlet\RestoreAzureSqlManagedDatabase.cs(24,35): error CS0103: The name 'PointInTimeRestoreFromNameAndResourceGroupParameterSet' does not exist in the current context [c:\workspace\powershell\src\ResourceManager\Sql\Commands.Sql\Commands.Sql.csproj]

**Pull request merged**